integration/src/main/java/com/aws/greengrass/mqtt/moquette/ClientDeviceAuthorizer.java

```diff
@@ -89,27 +100,39 @@ public boolean checkValid(String clientId, String username, byte[] password) {
     public boolean canWrite(Topic topic, String user, String client) {
         String resource = "mqtt:topic:" + topic;
         boolean canPerform = false;
+        String publishOp = "mqtt:publish";
         try {
-            canPerform = canDevicePerform(getOrCreateSessionForClient(client, user), "mqtt:publish", resource);
+            canPerform = canDevicePerform(getOrCreateSessionForClient(client, user), publishOp, resource);
         } catch (AuthenticationException e) {
             LOG.atWarn().cause(e).kv(CLIENT_ID, client).log("Unable to re-authenticate client.");
+            emitAuthErrorMetric(publishOp);
         }
         LOG.atDebug().kv("topic", topic).kv("isAllowed", canPerform).kv(CLIENT_ID, client)
             .log("MQTT publish request");
+
+        if (!canPerform) {
+            emitAuthErrorMetric(publishOp);
+        }
         return canPerform;
     }
 
     @Override
     public boolean canRead(Topic topic, String user, String client) {
         String resource = "mqtt:topicfilter:" + topic;
         boolean canPerform = false;
+        String subscribeOp = "mqtt:subscribe";
         try {
-            canPerform = canDevicePerform(getOrCreateSessionForClient(client, user), "mqtt:subscribe", resource);
+            canPerform = canDevicePerform(getOrCreateSessionForClient(client, user), subscribeOp, resource);
         } catch (AuthenticationException e) {
             LOG.atWarn().cause(e).kv(CLIENT_ID, client).log("Unable to re-authenticate client.");
+            emitAuthErrorMetric(subscribeOp);
```

Comment:
this will duplicate, since you do the same thing on L134

Reply:
fixed, thanks